File: 3. Arrays and String/747. Largest Number At Least Twice of Others/find_max.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        maxIdx = -1
        maxVal = -1

        for i in range(len(nums)):
            if nums[i] > maxVal:
                maxVal = nums[i]
                maxIdx = i

        for i in range(len(nums)):
            if 2 * nums[i] > maxVal and i != maxIdx:
                return -1

        return maxIdx

File: 3. Arrays and String/747. Largest Number At Least Twice of Others/find_max.cpp
class Solution
{
public:
    int dominantIndex(vector<int> &nums)
    {
        int maxIdx = -1;
        int maxVal = -1;

        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] > maxVal)
            {
                maxVal = nums[i];
                maxIdx = i;
            }
        }

        for (int i = 0; i < nums.size(); i++)
        {
            if (2 * nums[i] > maxVal && i != maxIdx)
            {
                return -1;
            }
        }
        return maxIdx;
    }
};